htcrystalball/collect.py

```diff
@@ -1,106 +1,48 @@
 """Retrieve, format, and store a system's condor slot configuration."""
 
-import json
-
 import htcondor
 
 from typing import Union
 
-from htcrystalball import SLOTS_CONFIGURATION
 from htcrystalball.utils import kib_to_gib, mib_to_gib
 
 
-def node_name_in_list(name: str, slots: list) -> Union[int, None]:
-    """Check if a node name is already in a list of slots."""
-    try:
-        return [slot['UtsnameNodename'] for slot in slots].index(name)
-    except ValueError:
-        return None
-
-
-def format_slots(slots: list) -> dict:
-    """
-    Reassign slot values to new keys for compatibility.
-
-    The dictionary of slots is formatted into a new dictionary similar to the
-    already used key:value pairs in HTCrystalball.
-    """
-    formatted = {"slots": []}
-
-    for slot in slots:
-        # TODO: Check if float conversion is necessary
-        name = slot['UtsnameNodename']
-        n_cpus = int(slot.get('TotalSlotCpus', 0))
-        n_gpus = int(slot.get('TotalSlotGPUs', 0))
-        n_slots = int(slot.get('TotalSlots', 0))
-        disk = float(slot.get('TotalSlotDisk', 0.0))
-        ram = float(slot.get('TotalSlotMemory', 0.0))
-        slot_type = slot['SlotType']
-
-        slot_size = {
-            "TotalSlotCpus": n_cpus,
-            "TotalSlotDisk": kib_to_gib(disk),
-            "TotalSlotMemory": mib_to_gib(ram)
-        }
-
-        if slot_type == "Partitionable":
-            if "gpu" in name and n_gpus != 0:
-                slot_size["SlotType"] = "gpu"
-                slot_size["TotalSlotGPUs"] = n_gpus
-            else:
-                slot_size["SlotType"] = "partitionable"
-                slot_size["TotalSlotGPUs"] = 0
-
-        elif slot_type == "Static":
-            slot_size["SlotType"] = "static"
-        else:
-            continue
-
-        slot_size["TotalSlots"] = n_slots
-        node_in_list = node_name_in_list(
-            slot["UtsnameNodename"],
-            formatted["slots"]
-        )
-
-        if node_in_list is not None:
-            if slot_size not in formatted["slots"][node_in_list]["slot_size"]:
-                formatted["slots"][node_in_list]["slot_size"].append(slot_size)
-        else:
-            formatted_slot = {
-                'UtsnameNodename': name,
-                'slot_size': [slot_size]
-            }
-            formatted["slots"].append(formatted_slot)
-
-    return formatted
-
-
 def collect_slots(filename: Union[str, None] = None) -> dict:
     """Get the condor config and creates a dict."""
-    status = {"slots": []}
-    projection = [
-        "SlotType", "UtsnameNodename", "Name", "TotalSlotCpus",
-        "TotalSlotDisk", "TotalSlotMemory", "TotalSlots", "TotalSlotGPUs"
-    ]
+    projection = ["SlotType", "UtsnameNodename", "TotalSlotCpus", "TotalSlotDisk", "TotalSlotMemory", "TotalSlots",
```

Comment:
This variable name is confusing to me, and it's not clear to me what it means in this context. I see that you're getting the name from the htcondor module later, but I don't know what /they/ mean either. ;-)

Can you explain what `projection` means here? I suggest either a more intuitive var name or a short comment.

Reply:
Ah okay, then maybe get_info_from_condor_pool

Reply:
`get_info_from_condor_pool` is an action, which is a better fit for a function. This is a variable (static), so it should be a noun rather than a verb.

Reply:
I think these are the fields that you want the htcondor pool to return for your query, yes? If so, a terse `fields` or `query` would make sense.  Alternatively `pool_query` or something like that.

Reply:
Projection is an SQL-based term, because HTCondor describes this argument as the `SELECT` statement in an SQL query:

> For those familiar with SQL queries, `requirements` performs the equivalent as the `WHERE` clause while `projection` performs the equivalent of the column listing in `SELECT`.

In SQL, `projection` means choosing which columns (or expressions) the query shall return.

Reply:
Ahh; ok, so it is a generic term, (albeit domain-specific). I'm fine with either `projection` here, or the more cross-domain `fields`.

Thanks for the explanation. :-)

Reply:
I went with query_data because the field describes the data we want to query.